common/flatpak-dir.c

```diff
@@ -13713,10 +13713,14 @@ local_match_prefix (FlatpakDir *self,
           if (!g_str_has_prefix (cur_parts[0], parts_prefix))
             continue;
 
-          g_ptr_array_add (matches, g_strconcat (parts[0], "/", partial_ref, NULL));
+          g_hash_table_add (matches, g_strconcat (parts[0], "/", partial_ref, NULL));
         }
     }
 
+  /* Also check deploys. In case remote-delete --force is run, we can end up
+   * with a deploy without a corresponding ref in the repo. */
+  flatpak_dir_collect_deployed_refs (self, parts[0], parts[1], parts[2], matches, NULL, NULL);
```

Comment:
Shouldn't this use parts_prefix instead of parts[1]? That is how the code above matches the prefix.

Reply:
Yeah, submitted https://github.com/flatpak/flatpak/pull/3092